src/enc/backward_references/hq.rs

```diff
@@ -209,16 +210,14 @@ impl<AllocF: Allocator<floatX>> ZopfliCostModel<AllocF> {
             num_bytes_: num_bytes,
             cost_cmd_: [0.0; 704],
             min_cost_cmd_: 0.0,
-            literal_costs_: if num_bytes.wrapping_add(2) > 0usize {
-                m.alloc_cell(num_bytes.wrapping_add(2))
-            } else {
-                AllocF::AllocatedMemory::default()
-            },
-            cost_dist_: if dist.alphabet_size > 0u32 {
-                m.alloc_cell(num_bytes.wrapping_add(dist.alphabet_size as usize))
-            } else {
-                AllocF::AllocatedMemory::default()
-            },
+            // FIXME: makes little sense to test if N+2 > 0 -- always true unless wrapping. Perhaps use allocate() instead?
+            literal_costs_: alloc_or_default::<floatX, _>(m, num_bytes + 2),
+            // FIXME: possible bug because allocation size is different from the condition
```

Comment:
This does seem to be a bug per https://chromium.googlesource.com/chromium/src/third_party/+/refs/heads/main/brotli/enc/backward_references_hq.c#109
but lets fix it in a subsequent PR and test it carefully